elements/pf-chip/pf-chip.ts

```diff
@@ -0,0 +1,121 @@
+import { LitElement, html, type PropertyValueMap } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import { query } from 'lit/decorators/query.js';
+import '@patternfly/elements/pf-button/pf-button.js';
+
+import styles from './pf-chip.css';
+
+export class ChipReadyEvent extends Event {
+  constructor() {
+    super('ready', { bubbles: true, composed: true });
+  }
+}
+export class ChipRemoveEvent extends Event {
+  constructor() {
+    super('remove', { bubbles: true, composed: true });
+  }
+}
+
+/**
+ * A **chip** is used to communicate a value or a set of attribute-value pairs within workflows that involve filtering a set of objects.
+ *
+ * @slot
+ *      chip text
+ *
+ * @csspart text - span container for chip text
+ *
+ * @fires { Event } click - when close button is clicked
+ */
+@customElement('pf-chip')
+export class PfChip extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * Accessible label for close button
+   */
+  @property({ attribute: 'close-label', type: String }) closeLabel = 'Close';
+
+  /**
+   * Flag indicating if chip is an overflow chip
+   */
+  @property({ reflect: true, attribute: 'overflow-chip', type: Boolean }) overflowChip = false;
+
+  /**
+   * Flag indicating if chip is read-only and cannot be removed
+   */
+  @property({ reflect: true, attribute: 'read-only', type: Boolean }) readOnly = false;
```

Comment:
```suggestion
  @property({ reflect: true, type: Boolean }) readonly = false;
```

https://github.com/patternfly/patternfly-elements/blob/a194ba7fcb035498683d70e6255b6f9441a9b3af/elements/pf-text-input/pf-text-input.ts#L189